### PR TITLE
WIP: create .phar using existing tools (composer, box.phar)

### PR DESCRIPTION
**Pull request by glensc**
#257

WIP: basic concept, when accepted needs cleanups (dropping library copies and unused scripts)

- [x] create phar using `box.phar`
- [ ] compression to use for phar (currently GZ)
- [ ] sign phar
- [ ] drop library copies
- [ ] drop unused scripts

INSTRUCTIONS:

commands must be run  in `dist/` subdir, git tree must be clean (uncommitted changes will not be "seen")
 - to make .phar, type `make`
 - to cleanup, type `make clean`

<del>
build system will attempt to find tools from `$PATH` fallbacking to current dir, if these are missing they can be downloaded with:
 - `make composer.phar`
 - `make box.phar`
</del>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	$(composer) install --no-dev
+
+%.phar: Makefile box.json composer.lock
+	$(php) -d phar.readonly=0 $(box) build -v
+
+box.phar:
+	curl -LSs https://box-project.github.io/box2/installer.php | php
+
+composer.phar:
+	curl -sS https://getcomposer.org/installer | php
```
Comment (defuse) [inline Makefile]: NACK on the above two curl commands. Let's instead document that developers need to install box.phar and composer in `docs/InternalDeveloperDocs`, and let them decide how securely they want to acquire them.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+TARGETS := defuse-crypto.phar
```
Comment (defuse) [inline Makefile]: Can we put things related to `.phar` building like this inside `other/`? I don't want regular users to get confused and think they need to run make. Building the `.phar` is something only this library's developers should do (everyone else should be using a signed `.phar`).

File: Makefile (hunk 1 above)
Comment (glensc) [inline Makefile]:
```
➔ PATH=/usr/bin make
Makefile:7: *** Required tool `box' not installed, see docs/InstallingAndVerifying.md.  Stop.
```

**Comment by glensc** [comment]
move the dist making scripts to dist/ subdir, also dropped automatic download and point use to documentation if tool can not be found from local system.

**Comment by defuse** [comment]
I incorporated this into #264, thanks!

**Pull request closed without merging**